database.cc

```diff
@@ -199,19 +199,22 @@ database::database(const db::config& cfg, database_config dbcfg)
     }))
     , _read_concurrency_sem(max_count_concurrent_reads,
         max_memory_concurrent_reads(),
+        "_read_concurrency_sem",
         max_inactive_queue_length(),
         [this] {
             ++_stats->sstable_read_queue_overloaded;
-            return std::make_exception_ptr(std::runtime_error("sstable read queue overloaded"));
+            return std::make_exception_ptr(std::runtime_error("_read_concurrency_sem: sstable read queue overloaded"));
```

Comment:
Just thinking out loud here: now that the semaphore has a name, maybe it could generate this error message on its own, there is no need to pass this factory method any-more.

Reply:
Please see my comment below